### ISSUE TITLE
QUANTS version 3.0.0.0 not splitting untrimmed and trimmed reads into separate files during primer trimming with cutadapt.

### ISSUE THREAD

**Issue by vaofford** [cancerit/QUANTS#18]
QUANTS version 3.0.0.0 

Primer trimming is not splitting untrimmed and trimmed reads into separate files (cutadapt).

The command for separating untrimmed reads is here:
https://github.com/cancerit/QUANTS/blob/40ae49e4808f7f2ebaeae26d7284b00ac746d42a/modules/local/cutadapt/main.nf#L35C25-L35C25

But is not being pulled through in the command options for cutadapt:
https://github.com/cancerit/QUANTS/blob/40ae49e4808f7f2ebaeae26d7284b00ac746d42a/modules/local/cutadapt/main.nf#L43

```
script:
    def software       = getSoftwareName(task.process)
    def prefix         = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
    def trimmed        = meta.single_end ? "-o ${prefix}_trimmed.fastq.gz" : "-o ${prefix}_trimmed_1.fastq.gz -p ${prefix}_trimmed_2.fastq.gz"
    def untrimmed_cmd  = meta.single_end ? "--untrimmed-output ${prefix}_untrimmed.fastq.gz" : "--untrimmed-output ${prefix}_untrimmed_1.fastq.gz --untrimmed-paired-output ${prefix}_untrimmed_2.fastq.gz"
    def untrimmed      = options.untrimmed ? untrimmed_cmd : ''
    """
    cutadapt \\
        --cores $task.cpus \\
        --json=${prefix}.cutadapt.json \\
        $options.args \\
        $trimmed \\
        $untrimmed \\
        $reads \\
        > ${prefix}.cutadapt.log
    echo \$(cutadapt --version) > ${software}.version.txt
    """
```

See cutadapt command being run in cutadapt log:

Adapter trimming:
```
Command line parameters: --cores 4 --json=SGE_1121590538_Rep1_CMDC1_Day0.adapter.cutadapt.json -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -o SGE_1121590538_Rep1_CMDC1_Day0.adapter_trimmed.fastq.gz 46503#47.fq.gz
```

Primer trimming:
```
Command line parameters: --cores 4 --json=SGE_1121590538_Rep1_CMDC1_Day0.primer.cutadapt.json -g GGCAGGATGGATTCTGAGCAGAGCC...TTCAGTGTGCCTGAAAGTGGTTTGC -o SGE_1121590538_Rep1_CMDC1_Day0.primer_trimmed.fastq.gz SGE_1121590538_Rep1_CMDC1_Day0.adapter_trimmed.fastq.gz
```

Options are defined in:
https://github.com/cancerit/QUANTS/blob/40ae49e4808f7f2ebaeae26d7284b00ac746d42a/conf/modules.config#L35

**Comment by vaofford** [comment]
Expected primer trimming command:

```
Command line parameters: --cores 4 --json=SGE_1121590538_Rep1_CMDC1_Day0.primer.cutadapt.json -g GGCAGGATGGATTCTGAGCAGAGCC...TTCAGTGTGCCTGAAAGTGGTTTGC -o SGE_1121590538_Rep1_CMDC1_Day0.primer_trimmed.fastq.gz --untrimmed-output SGE_1121590538_Rep1_CMDC1_Day0.primer_untrimmed.fastq.gz SGE_1121590538_Rep1_CMDC1_Day0.adapter_trimmed.fastq.gz
```

**Comment by y-popov** [comment]
This is embarassing, I believe that worked at some point.

### 
TLDR
The easiest way to fix that is to change this line
```nextflow
def untrimmed      = options.untrimmed ? untrimmed_cmd : ''
```
to
```nextflow
def untrimmed      = params.options.untrimmed ? untrimmed_cmd : ''
```

### Explanation
When `PRIMER_TRIMMING` workflow passes options from the config to a process
```nextflow
include { CUTADAPT as CUTADAPT_PRIMER  } from '../../modules/local/cutadapt/main' addParams( options: primer_cutadapt_options )
```
they land (being appended) into `params.options` variable.  This variable is used to create another variable `options` with function `initOptions` however this function generates an object with a predefined structure that lacks custom field `untrimmed` and so ignores it. This is why we lose it. As suggested above, we can use `params.options.untrimmed` directly.
